### DISCUSSION TITLE
How to implement custom interceptor?

### DISCUSSION THREAD

**Discussion by FahriDevZ** [jmcdo29/ogma#1571] [General]
I have module with [@grammyjs/nestjs](https://github.com/grammyjs/nestjs/tree/main/packages/grammy-nestjs)
when i test send message i always got error from `this.delegate.setRequestId(context, correlationId);` [interceptor/ogma.interceptor.ts#L45](https://github.com/jmcdo29/ogma/blob/46dc3678778afb1be5f9bfd15322a32f2bbd1578/packages/nestjs-module/src/interceptor/ogma.interceptor.ts#LL45C5-L45C56) is because from ExecutionContextHost.contextType is  `grammy` and no grammy exits in DelegatorService#getParser [interceptor/providers/delegator.service.ts#L21](https://github.com/jmcdo29/ogma/blob/46dc3678778afb1be5f9bfd15322a32f2bbd1578/packages/nestjs-module/src/interceptor/providers/delegator.service.ts#L44).

**Comment by jmcdo29** [comment]
Ah, this is something I'm fixing in `@ogma/nestjs-module@5.0.0` to make it more flexible. There's an issue with `necord` as well for this same reason. In the meantime, I'll probably need to add a skip over the interceptor if the type isn't a known one until v5 is released. I'm waiting for a PR in the Nest repo to add some functionality to websockets before I go on with this update as well
